classes/BookingManager.php

```diff
@@ -187,9 +185,10 @@ protected function filterNextBookableTable($tables, int $noOfGuests)
         foreach ($tables as $table) {
             if ($table->is_joinable) {
                 $previousCapacity += $table->max_capacity;
-                $result->push($table);
-                if ($previousCapacity >= $noOfGuests)
+                if ($previousCapacity >= $noOfGuests) {
+                    $result->push($table);
                     break;
```

Comment:
This looks like it would only join two tables, unlike the previous approach which joins the tables until it can accommodate guests.

Reply:
Got you, i had missed that multiple tables were being assigned to the reservation, thought it was just the last one.

Updated the logic and also handled the case where one table can handle the entire booking but isnt joinable being checked after the joinable tables (reset $result)

Reply:
Looks like it now 👍